### PR TITLE
chore: properly render mkdocs.yml

### PR DESCRIPTION
**Pull request by mmorel-35**
Part 2, closes #1519

### CODE REVIEW

File: modulegen/internal/dependabot/writer.go [hunk 1]
```diff
@@ -2,16 +2,21 @@ package dependabot
...
+	}
 	data, err := yaml.Marshal(config)
 	if err != nil {
 		return err
 	}
-	header := "# This file is autogenerated by the modulegen code generator. Please look at the generator code when updating it.\n"
+	header := "# This file is autogenerated by the 'modulegen' tool.\n# Please look at the generator code when updating it.\n"
 	data = append([]byte(header), data...)
-	return os.WriteFile(configFile, data, 0o777)
+	return os.WriteFile(configFile, data, 0o644)
```
Comment (mdelapenya) [inline modulegen/internal/dependabot/writer.go:21]: Could you explain the file permission change?

File: modulegen/internal/mkdocs/writer.go [hunk 2]
```diff
@@ -2,14 +2,35 @@ package mkdocs
...
+	err := os.MkdirAll(filepath.Dir(configFile), 0o755)
+	if err != nil {
+		return err
+	}
 	data, err := yaml.Marshal(config)
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(configFile, data, 0o777)
+	return os.WriteFile(configFile, overrideData(data), 0o644)
```
Comment (mdelapenya) [inline modulegen/internal/mkdocs/writer.go:20]: I like this override more than in the original PR. Neat!

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks!!!

File: modulegen/internal/dependabot/writer.go (hunk 1 above)
Comment (mmorel-35) [reply modulegen/internal/dependabot/writer.go:21]: I tried to fit git repository folder (755) and file(644) permissions . 777 seems to be to much. But maybe I’m mising something ?

Comment (mdelapenya) [reply modulegen/internal/dependabot/writer.go:21]: No no, I'm just trying to understand the change. I think it's probably fine to reduce the number

**Comment by mdelapenya** [comment]
Merging, as the CI already passed. Thanks!

**Pull request merged**